src/input/keyboard/mod.rs

```diff
@@ -751,6 +751,28 @@ impl<D: SeatHandler + 'static> KeyboardHandle<D> {
     ) -> Option<T>
     where
         F: FnOnce(&mut D, &ModifiersState, KeysymHandle<'_>) -> FilterResult<T>,
+    {
+        let (filter_result, mods_changed) = self.intercept(data, keycode, state, filter);
+        if let FilterResult::Intercept(val) = filter_result {
+            // the filter returned false, we do not forward to client
```

Comment:
If we are moving this comment around we might as well fix it while at it, filter no longer returns bool so `the filter returned false` is outdated.